### ISSUE TITLE
Refactor `/search/chat_channels` endpoint

### ISSUE THREAD

**Issue by atsmith813** [forem/forem#13277]
In https://github.com/forem/forem/pull/13235 we refactored how `/search/chat_channels` works by essentially removing "search" functionality and turning it into more of a "filter". [As noted in the PR](https://github.com/forem/forem/pull/13235#discussion_r606368287):
> You'll notice I left everything in the Search namespace (the route, the controller, and the service naming). I originally started removing chat_channels from search all together since this behaves more like a filter than a search.
>
> However, I discovered there are some JS components that are using this endpoint as an "index". I started refactoring everything, but it quickly got out of hand and out of scope for what this PR is intended to address. I also realized it may actually be nice to leave everything in the search namespace so that when we re-do the search for Connect in the future, it's a smoother experience.
>
> With that said, since we have the ChatChannelsController and the ChatChannelMembershipsController each with index actions, I think the right move would be to move those JS components to them so we don't have multiple controllers performing the same actions.

This issue is to essentially update all the JS components on the frontend to use a newly refactored endpoint that lives in a more appropriate spot (i.e. `ChatChannelMembershipsController` or `ChatChannelsController`) than the `SearchController`. Once this issue is complete, the `chat_channels` action should no longer exist in the `SearchController` and instead, a new endpoint should be able to take care of any "filtering" and "indexing" being done by the JS components.

**Comment by rhymes** [comment]
+1 on removing the code out of search and into a dedicated query object / service

**Comment by Rafi993** [comment]
Hi, if this PR is not worked on I would like to help.

**Comment by cmgorton** [comment]
> Hi, if this issue is not worked on I would like to help.

Thanks Rafi! Just assigned you to it.

**Comment by Rafi993** [comment]
Sorry I did not look at this issue for a while I will look at this over next week but in the mean time if this is a urgent issue and I am blocking it do feel free to change the owner.

**Comment by cmgorton** [comment]
I don't believe it is urgent so if you still want to work on it that is totally fine. If not we can remove your assignment. Just let me know what works for you Rafi 😄

**Comment by Rafi993** [comment]
Thank you @cmgorton. I'll actually un-assign myself now and comeback later to pick up this issue if this still open :)

**Comment by twinsfan421** [comment]
I can work on this

**Comment by cmgorton** [comment]
> I can work on this

Thank you!

**Comment by cmgorton** [comment]
Hey @twinsfan421 
I was reminded by a teammate that we are putting work on Connect on hold at the moment while we plan out the future of Connect and any work we want done on it. I don't want you to work on this issue only to have us remove the code later on, so I am going to unassign you for now. Please let me know if there are any other issues you would like to work on and thanks again for your willingness to contribute here🙏

**Comment by cmgorton** [comment]
I am going to close this issue since we are moving to deprecate connect. 
[Deprecating Connect Chat](https://forem.dev/foremteam/deprecating-connect-chat-2kgi)
